sharding/notary/notary.go

```diff
@@ -14,69 +12,34 @@ import (
 	shardparams "github.com/ethereum/go-ethereum/sharding/params"
 )
 
-// SubscribeBlockHeaders checks incoming block headers and determines if
-// we are an eligible notary for collations. Then, it finds the pending tx's
-// from the running geth node and sorts them by descending order of gas price,
-// eliminates those that ask for too much gas, and routes them over
-// to the SMC to create a collation.
-func subscribeBlockHeaders(client mainchain.Client) error {
-	headerChan := make(chan *types.Header, 16)
-
-	_, err := client.ChainReader().SubscribeNewHead(context.Background(), headerChan)
-	if err != nil {
-		return fmt.Errorf("unable to subscribe to incoming headers. %v", err)
-	}
-
-	log.Info("Listening for new headers...")
-
-	for {
-		// TODO: Error handling for getting disconnected from the client.
-		head := <-headerChan
-		// Query the current state to see if we are an eligible notary.
-		log.Info(fmt.Sprintf("Received new header: %v", head.Number.String()))
-
-		// Check if we are in the notary pool before checking if we are an eligible notary.
-		v, err := isAccountInNotaryPool(client)
-		if err != nil {
-			return fmt.Errorf("unable to verify client in notary pool. %v", err)
-		}
-
-		if v {
-			if err := checkSMCForNotary(client, head); err != nil {
-				return fmt.Errorf("unable to watch shards. %v", err)
-			}
-		}
-	}
-}
-
 // checkSMCForNotary checks if we are an eligible notary for
 // collation for the available shards in the SMC. The function calls
 // getEligibleNotary from the SMC and notary a collation if
 // conditions are met.
-func checkSMCForNotary(client mainchain.Client, head *types.Header) error {
+func checkSMCForNotary(caller mainchain.ContractCaller, head *types.Header) error {
```

Comment:
Why not pass in SMCCaller and the account as a seperate argument (since it doesnt seem to be related to the contract at all)?